autoware_lanelet2_map_validator/src/validators/lane/lanelet_geometry.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lanelet2_map_validator/validators/lane/lanelet_geometry.hpp"
+
+#include "lanelet2_map_validator/utils.hpp"
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+namespace lanelet::autoware::validation
+{
+namespace
+{
+lanelet::validation::RegisterMapValidator<LaneletGeometryValidator> reg;
+}
+
+lanelet::validation::Issues LaneletGeometryValidator::operator()(const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  lanelet::autoware::validation::appendIssues(issues, check_lanelet_geometry(map));
+
+  return issues;
+}
+
+lanelet::validation::Issues LaneletGeometryValidator::check_lanelet_geometry(
+  const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  for (const auto & lanelet : map.laneletLayer) {
+    const auto & left_bound = lanelet.leftBound();
+    const auto & right_bound = lanelet.rightBound();
```

Comment:
```suggestion
    const auto left_bound = lanelet.leftBound();
    const auto right_bound = lanelet.rightBound();
```

We don't need them to be references to leftBound and rightBound